### PR TITLE
VPN-5782 - Add tests for DaemonAccessControl

### PR DESCRIPTION
**Pull request by brizental**

### CODE REVIEW

File: tests/unit_tests/CMakeLists.txt [hunk 1]
```diff
@@ -122,10 +123,14 @@ target_sources(app_unit_tests PRIVATE
     testthemes.h
     testurlopener.cpp
     testurlopener.h
+    testdaemonaccesscontrol.cpp
```
Comment (vinocher) [inline tests/unit_tests/CMakeLists.txt:126]: Should these be added in the existing, sorted order?

File: src/daemon/daemonaccesscontrol.h [hunk 2]
```diff
@@ -29,6 +29,10 @@ class DaemonAccessControl final {
   bool isCommandAuthorizedForPeer(const QString& command,
                                   const QLocalSocket* socket);
 
+#ifdef UNIT_TEST
+  static inline void setPeerId(int peerId) { s_mockPeerId = peerId; }
```
Comment (vinocher) [inline src/daemon/daemonaccesscontrol.h:33]: Nit: Should this be named SetMockPeerId?

File: tests/unit_tests/testdaemonaccesscontrol.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+void TestDaemonAccessControl::testInactiveCommands() {
+  auto dac = DaemonAccessControl::instance();
+
+  QVERIFY(dac->isCommandAuthorizedForPeer("status", nullptr));
+  QVERIFY(!dac->isCommandAuthorizedForPeer("logs", nullptr));
+  QVERIFY(!dac->isCommandAuthorizedForPeer("cleanlogs", nullptr));
+  QVERIFY(!dac->isCommandAuthorizedForPeer("deactivate", nullptr));
+
+  // A peer id of -1 means activation failed,
+  // access crontrol defaults to unauthorized.
```
Comment (vinocher) [inline tests/unit_tests/testdaemonaccesscontrol.cpp:23]: Nit: typo for "control"

File: tests/unit_tests/testdaemonaccesscontrol.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  dac->setPeerId(-1);
+  QVERIFY(!dac->isCommandAuthorizedForPeer("activate", nullptr));
+  // Set peer id to something valid for activateion to succed.
```
Comment (vinocher) [inline tests/unit_tests/testdaemonaccesscontrol.cpp:26]: Nit: Typos

File: tests/unit_tests/testdaemonaccesscontrol.cpp [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  QVERIFY(dac->isCommandAuthorizedForPeer("cleanlogs", nullptr));
+  QVERIFY(dac->isCommandAuthorizedForPeer("deactivate", nullptr));
+}
+
+void TestDaemonAccessControl::testActiveAuthorizedPeer() {
+  auto dac = DaemonAccessControl::instance();
+
+  // Set the peer id to a normal user peer id and activate.
+  // That will make `1` the session owner.
+  dac->setPeerId(1);
```
Comment (vinocher) [inline tests/unit_tests/testdaemonaccesscontrol.cpp:53]: Consider calling dac->StartSession() after setPeerId, here and elsewhere in this file, to test that path.

File: tests/unit_tests/testdaemonaccesscontrol.h [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+class TestDaemonAccessControl final : public TestHelper {
+  Q_OBJECT
+
+ private slots:
+  void initTestCase();
+  /**
+   * * "status" command is allowed for any peer when inactive
+   * * "activate" command is allowed for any peer when inactive
+   * * commands other than "activate" or "status" are not allowed
+   * is allowed for any peer when inactive.
```
Comment (vinocher) [inline tests/unit_tests/testdaemonaccesscontrol.h:16]: Does line 16 need to be removed?

File: tests/unit_tests/testdaemonaccesscontrol.h [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+   *
+   */
+  void testInactiveCommands();
```
Comment (vinocher) [inline tests/unit_tests/testdaemonaccesscontrol.h:19]: Consider if names similar to testCommandsWhenInactive(), testAuthorizedPeerWhenActive() etc. are an improvement.

File: tests/unit_tests/testdaemonaccesscontrol.cpp [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+  QVERIFY(!dac->isCommandAuthorizedForPeer("cleanlogs", nullptr));
+  QVERIFY(!dac->isCommandAuthorizedForPeer("deactivate", nullptr));
+
+  // A peer id of -1 means activation failed,
+  // access crontrol defaults to unauthorized.
+  dac->setPeerId(-1);
+  QVERIFY(!dac->isCommandAuthorizedForPeer("activate", nullptr));
+  // Set peer id to something valid for activateion to succed.
+  dac->setPeerId(1);
+  QVERIFY(dac->isCommandAuthorizedForPeer("activate", nullptr));
```
Comment (vinocher) [inline tests/unit_tests/testdaemonaccesscontrol.cpp:28]: Consider adding a couple of lines to verify that "status" is authorized for any peer when the VPN is inactive.

File: tests/unit_tests/testdaemonaccesscontrol.cpp (hunk 5 above)
Comment (brizental) [reply tests/unit_tests/testdaemonaccesscontrol.cpp:53]: `startSession` is private API, that is called internally whenever the `activate` command is parsed.

**Review by vinocher — APPROVED** [review]
Looks good!

**Pull request merged**